### ISSUE TITLE
Cannot retrieve loss after /  during training

### ISSUE THREAD

**Issue by enavarroAI** [ncullen93/torchsample#75]
Hey guys,
While training and during the epoch, I can see the train loss value changing, but once the epoch is over, the value is 'deleted', as in it's shown as an empty list.
I tried using CSVLogger and History callbacks, and both retrieve empty lists for train_loss and val_loss.
I ran it in my code, and also in the MNIST example included in this repository.

Anyone with this issue?
Thanks
![capture](https://user-images.githubusercontent.com/27867788/33195694-7768885c-d09f-11e7-9cdc-19cc89722184.PNG)

**Comment by ririw** [comment]
So the issue can be found [here](https://github.com/ncullen93/torchsample/blob/master/torchsample/callbacks.py#L204)

A simple fix: replace the whole function body with:

    def on_epoch_end(self, epoch, logs=None):
         self.epoch_metrics.update(self.batch_metrics)

However I don't know enough about the callbacks and so on to be sure that's the "correct" approach.

**Comment by justiceamoh** [comment]
I am having the same issue. Using your (@ririw) simple fix works for training loss, but not validation. The other callbacks don't work also.

**Comment by justiceamoh** [comment]
I've been working on this today and I think I have come up with a fix that ensures the other callbacks work also. I will create a PR later in the day. Stay tuned.
